packages/web/src/services/audius-backend/Tipping.ts

```diff
@@ -58,3 +61,71 @@ export const fetchSupporters = async ({
     return []
   }
 }
+
+export type UserTipRequest = {
+  userId?: string
+  limit?: number
+  offset?: number
+  receiverMinFollowers?: number
+  receiverIsVerified?: boolean
+  currentUserFollows?: 'sender' | 'receiver' | 'sender_or_receiver'
+  uniqueBy?: 'sender' | 'receiver'
+  minSlot?: number
+  maxSlot?: number
+  txSignatures?: string[]
+}
+type UserTipOmitIds = 'sender_id' | 'receiver_id' | 'followee_supporter_ids'
+type UserTipResponse = Omit<UserTip, UserTipOmitIds> & {
+  sender: any
+  receiver: any
+  followee_supporters: any[]
+}
+
+export const fetchRecentUserTips = async ({
+  userId,
+  limit,
+  offset,
+  receiverMinFollowers,
+  receiverIsVerified,
+  currentUserFollows,
+  uniqueBy,
+  minSlot,
+  maxSlot,
+  txSignatures
+}: UserTipRequest): Promise<UserTipResponse[]> => {
+  try {
+    await waitForLibsInit()
+    const response = await libs().discoveryProvider._makeRequest({
+      endpoint: `/v1/full/tips`,
+      queryParams: {
+        user_id: userId,
```

Comment:
would it make sense to create queryParams const and then use it here and also for console.error?

Reply:
yup makes sense